x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/QueryParam.java

```diff
@@ -6,34 +6,24 @@
  */
 package org.elasticsearch.xpack.esql.parser;
 
+import org.elasticsearch.xpack.esql.core.type.DataType;
+
 import java.util.Objects;
 
 /**
  * Represent a strongly typed parameter value
  */
-public class TypedParamValue {
+public class QueryParam {
 
-    public final Object value;
-    public final String type;
-    private boolean hasExplicitType;        // the type is explicitly set in the request or inferred by the parser
+    private final String name;
+    private final Object value;
+    private final DataType type;
     private ContentLocation tokenLocation; // location of the token failing the parsing rules
 
-    public TypedParamValue(String type, Object value) {
-        this(type, value, true);
-    }
-
-    public TypedParamValue(String type, Object value, boolean hasExplicitType) {
+    public QueryParam(String name, Object value, DataType type) {
+        this.name = name;
         this.value = value;
         this.type = type;
-        this.hasExplicitType = hasExplicitType;
-    }
-
-    public boolean hasExplicitType() {
-        return hasExplicitType;
-    }
-
-    public void hasExplicitType(boolean hasExplicitType) {
-        this.hasExplicitType = hasExplicitType;
     }
 
     public ContentLocation tokenLocation() {
```

Comment:
This method is not used anymore, but maybe it should to improve the error messaging.

Reply:
This token location is not set for each parameter yet, probably we should set it for each parameter, so that the location can be printed in the error messages.